### DISCUSSION TITLE
deleteMany not working on prisma 6.1.0

### DISCUSSION THREAD

**Discussion by alzatius** [prisma/prisma#25912] [Q&A]
### Bug description

The deleteMany directive (when inserting or updating records) does not work on prisma 6.1.0. The bug was detected after upgrading. A downgrade to 5.22.0 solves the issue.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```
6.1.0

**Comment by ludralph** [comment]
Hi @alzatius 

Can you please provide the relevant parts of your schema and query? This will help in reproducing the issue.

**Comment by alzatius** [comment]
Hi Rafael,

 

The relevant code is:

 
```ts
export const updateBoard = async (board: QBoard) => {

  let retData: QData = {

    isError: false,

    isShop: false,

    errorMsg: "",

    data: null,

  };

  let itemModels: MediaItemBoard[] = [];

  board.items.map((item, index) => {

    itemModels.push({

      index: index + 1,

      fileId: item.fileId,

      fileExt: item.fileExt,

      alt: item.alt,

    } as MediaItemBoard);

  });

  let updBoard = null;

  try {

    updBoard = await prisma.mediaPanelBoard.update({

      where: {

        id: board.panelId,

      },

      data: {

        name: board.name,

        itemHeight: board.itemHeight,

        itemWidth: board.itemWidth,

        items: {

          deleteMany: {},

          create: itemModels,

        },

      },

      include: {

        items: true,

      },

    });

  } catch (e) {

    retData.isError = true;

    if (e instanceof Prisma.PrismaClientKnownRequestError) {

      if (e.code === "P2025") {

        // the code for non-found record

        return await addAtelierBoard(board);

      }

    } else if (

      e instanceof Prisma.PrismaClientKnownRequestError ||

      e instanceof Prisma.PrismaClientUnknownRequestError

    ) {

      retData.errorMsg = e.message;

    } else {

      retData.errorMsg = "Unknown message";

    }

  } finally {

    retData.data = updBoard;

  }

  return retData;

};
```
 

And the schema:

 
```prisma
model MediaPanelBoard {

  id         String           @id @unique

  createdAt  DateTime         @default(now())

  updatedAt  DateTime         @updatedAt

  name       String           @default("")

  atelierId  String           @default("")

  itemHeight Int              @default(100)

  itemWidth  Int              @default(100)

  atelier    UserAtelier      @relation(fields: [atelierId], references: [id], onDelete: Cascade)

  items      MediaItemBoard[]

 

  @@index([atelierId])

}

 

model MediaItemBoard {

  id        String          @id @default(uuid())

  createdAt DateTime        @default(now())

  updatedAt DateTime        @updatedAt

  boardId   String          @default("")

  index     Int             @default(1)

  fileId    String          @default("")

  fileExt   String          @default("")

  alt       String          @default("")

  board     MediaPanelBoard @relation(fields: [boardId], references: [id], onDelete: Cascade)

 

  @@index([boardId])

}
```
 

 

The issue started to happen after I upgraded from 5.22 to 6.1.

 

I hope this helps.

 

Cristian

**Comment by alzatius** [comment]
Hi again,

 

I forgot to include the schema’s config:

 

generator client {

  provider = "prisma-client-js"

}

 

datasource db {

  provider     = "mysql"

  url          = env("DATABASE_URL")

  relationMode = "prisma"

}

 

 

Cheers,

Cristian

**Comment by ludralph** [comment]
Hi @alzatius 

Can you please share the error message you got? I can confirm the `deleteMany` works with Prisma version 6.1.0.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
